File: app/src/main/java/org/dhis2/data/tuples/Single.java
package org.dhis2.data.tuples;

import androidx.annotation.NonNull;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class Single<A> {

    @NonNull
    public abstract A val0();

    @NonNull
    public static <A> Single<A> create(@NonNull A val) {
        return new AutoValue_Single<>(val);
    }
}

File: app/src/main/java/org/dhis2/data/tuples/Single.kt
package org.dhis2.data.tuples

import com.google.auto.value.AutoValue

@AutoValue
abstract class Single<A> {

    abstract fun val0(): A

    companion object {

        fun <A> create(`val`: A): Single<A> {
            return AutoValue_Single(`val`)
        }
    }
}
